libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -32,6 +32,41 @@ using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
+{
+
+void copyMissingTags(StructurallyDocumentedAnnotation& _target, CallableDeclarationAnnotation const& _source)
+{
+	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>(_source);
+
+	set<string> existingTags;
+
+	for (auto const& iterator: _target.docTags)
```

Comment:
`_target.docTags` is a `multimap` - do we really need another set for the existing keys?

Reply:
Exactly because it is a multimap we do need this. If we add a multi-value that didn't previously exist to `_target`, we would not add the remaining values of that entry if we just check for existence of the key.

Reply:
ah, right! Thanks for the explanation!